src/whitenoise/event_processor/event_handlers/handle_contact_list.rs

```diff
@@ -30,72 +30,57 @@ impl Whitenoise {
+            .get_latest_follow_list_timestamp(&self.database)
+            .await?;
+        let event_timestamp = event.created_at.as_u64() * 1000; // Convert seconds to milliseconds
+
+        if let Some(current_time) = current_event_time {
+            if event_timestamp <= current_time {
+                tracing::debug!(
+                    target: "whitenoise::handle_contact_list",
+                    "Ignoring older contact list event (event: {}, current: {}) for account {}",
+                    event_timestamp,
+                    current_time,
+                    account.pubkey.to_hex()
+                );
+                return Ok(());
+            }
+        }
 
         tracing::debug!(
-            target: "whitenoise::handle_contact_list_internal",
-            "Processing contact list for account {}: {} to follow, {} to unfollow",
-            account.pubkey.to_hex(),
-            users_to_follow.len(),
-            users_to_unfollow.len()
+            target: "whitenoise::handle_contact_list",
+            "Processing contact list event (timestamp: {}) for account {}",
+            event_timestamp,
+            account.pubkey.to_hex()
         );
 
-        // Check if we have changes to make before processing
-        let has_changes = !users_to_follow.is_empty() || !users_to_unfollow.is_empty();
-
-        if !has_changes {
-            tracing::debug!(
-                target: "whitenoise::handle_contact_list_internal",
-                "No changes to make to contact list for account {}",
-                account.pubkey.to_hex()
-            );
-            return Ok(());
-        }
+        let contacts_from_event = NostrManager::pubkeys_from_event(&event);
 
-        for pubkey in &users_to_follow {
-            let (user, newly_created) =
-                crate::whitenoise::users::User::find_or_create_by_pubkey(pubkey, &self.database)
-                    .await?;
+        // Use the new bulk update method and get the list of newly created users
+        let newly_created_pubkeys = account
+            .update_follows_from_event(contacts_from_event.clone(), event_timestamp, &self.database)
+            .await?;
 
-            if newly_created {
-                self.background_fetch_user_data(&user).await?;
-            }
-
-            account.follow_user(&user, &self.database).await?;
-        }
+        // Store count for logging before consuming the vector
+        let newly_created_count = newly_created_pubkeys.len();
 
-        for pubkey in &users_to_unfollow {
-            let (user, created) =
-                crate::whitenoise::users::User::find_or_create_by_pubkey(pubkey, &self.database)
-                    .await?;
-            if created {
+        // Background fetch user data for newly created users
+        for pubkey in newly_created_pubkeys {
+            if let Ok((user, _)) =
+                crate::whitenoise::users::User::find_or_create_by_pubkey(&pubkey, &self.database)
```

Comment:
It's annoying having to do find_or_create_by_pubkey again, even though we basically just did it within `update_follows_from_event` but couldn't find a clean way to make it better so I guess it's fine at least for now